src/node/Launcher.ts

```diff
@@ -392,63 +324,6 @@ class FirefoxLauncher implements ProductLauncher {
     }
   }
 
-  async connect(
-    options: BrowserOptions & {
-      browserWSEndpoint?: string;
-      browserURL?: string;
-      transport?: ConnectionTransport;
-    }
-  ): Promise<Browser> {
-    const {
-      browserWSEndpoint,
-      browserURL,
-      ignoreHTTPSErrors = false,
-      defaultViewport = { width: 800, height: 600 },
-      transport,
-      slowMo = 0,
-    } = options;
-
-    assert(
-      Number(!!browserWSEndpoint) +
-        Number(!!browserURL) +
-        Number(!!transport) ===
-        1,
-      'Exactly one of browserWSEndpoint, browserURL or transport must be passed to puppeteer.connect'
-    );
-
-    let connection = null;
-    if (transport) {
-      connection = new Connection('', transport, slowMo);
-    } else if (browserWSEndpoint) {
-      const connectionTransport = await WebSocketTransport.create(
-        browserWSEndpoint
-      );
-      connection = new Connection(
-        browserWSEndpoint,
-        connectionTransport,
-        slowMo
-      );
-    } else if (browserURL) {
-      const connectionURL = await getWSEndpoint(browserURL);
-      const connectionTransport = await WebSocketTransport.create(
-        connectionURL
-      );
-      connection = new Connection(connectionURL, connectionTransport, slowMo);
-    }
-
-    const { browserContextIds } = await connection.send(
-      'Target.getBrowserContexts'
-    );
-    return Browser.create(
-      connection,
-      browserContextIds,
-      ignoreHTTPSErrors,
-      defaultViewport,
-      null,
-      () => connection.send('Browser.close').catch(debugError)
-    );
-  }
```

Comment:
The `FirefoxLauncher` class and the `ChromeLauncher` class both defined the `connect` method, but it was identical on both. Hence now there being just a single function to do the connecting.